### ISSUE TITLE
daily_dqn feature scaler is never persisted, so live predictions normalize on a different window than training

### ISSUE THREAD

**Issue by seansorek** [seansorek/trading_sim#123]
## Description

The DQN is trained on z-scored features, but the scaler that produced those z-scores is thrown away at save time and re-derived from different data at predict time.

**Training** — `TradingEnv.__init__` fits a per-symbol scaler on a warmup window and applies it in place:

```python
# rl_env.py:54-60
fit_end = min(252, max(self.window + 1, len(self.df) // 2))
self.scaler = {}
for c in self.features:
    mu = float(self.df[c].iloc[:fit_end].mean())
    sd = float(self.df[c].iloc[:fit_end].std() or 1.0)
    self.scaler[c] = (mu, sd)
```

**Saving** — `DQNAgent.save` (`dqn_agent.py:245`) writes only `{"model": self.q.state_dict(), ...}` plus hyperparameters. `self.scaler` lives on the env, never on the agent, and never reaches `models/dqn_agent.pt`.

**Prediction** — `predict_symbol` re-derives mu/sd from scratch using the same *formula* on a completely different frame:

```python
# predict_next_day_lite.py:558-566
fit_end = min(252, max(dqn_window + 1, n_rows // 2))
for ci, col in enumerate(FEATURE_COLS):
    mu = float(X_all[:fit_end, ci].mean())
    sd = float(X_all[:fit_end, ci].std() or 1.0)
    X_normed[:, ci] = (X_all[:, ci] - mu) / sd
```

`X_all` here is the last `cfg.data.history_days` (~1000 calendar days) of bars ending today, so `fit_end` selects the oldest ~252 bars of a rolling window that moves forward every single day. `train_dqn` builds its envs from a fixed `start`/`end` (`train_models.py:471`), so its warmup window is anchored to the training period. The two windows cover different calendar periods and therefore different mu/sd — and the live window's anchor drifts daily.

**Location:** [rl_env.py:50-60](https://github.com/seansorek/trading_sim/blob/main/rl_env.py#L50-L60), [dqn_agent.py:245-252](https://github.com/seansorek/trading_sim/blob/main/dqn_agent.py#L245-L252), [predict_next_day_lite.py:556-566](https://github.com/seansorek/trading_sim/blob/main/predict_next_day_lite.py#L556-L566)

## Second, independent skew in the same path

`TradingEnv` calls `make_daily_features(raw)` with **no** `spy_df` (`rl_env.py:34`). With `spy_df=None` the market-relative features are set to a constant:

```python
# daily_features.py:146-148
feats["ret_1d_vs_spy"] = 0.0
feats["ret_5d_vs_spy"] = 0.0
```

and the `_ZSCORE_FEATURES` pass leaves them at 0. So the DQN is trained with `ret_1d_vs_spy` and `ret_5d_vs_spy` identically zero across every symbol and every bar. At predict time `predict_symbol` passes a real `spy_df`, so those two of the 30 features arrive as live z-scores the network has never seen take a nonzero value.

## Impact

`daily_dqn` is live: `models/dqn_agent.pt` is committed, and `load_models` loads it unconditionally whenever the file exists (independent of `prediction.models` in `config/default.yaml`), so its signals go into `tomorrow_trades.json`, `predictions/history.jsonl`, the Discord post, and the consensus block that flags symbols where ≥2 models agree.

Both skews mean the Q-network is evaluated on inputs drawn from a different distribution than it was trained on. Its Q-values — and therefore the BUY/SELL/HOLD gate in `gate_dqn_signal` and the reported confidence — are not the ones the training run validated. Unlike a scaling bug that shows up as an obvious crash, this degrades silently: the signals still look well-formed. The daily drift of the live warmup anchor also means the same model can emit different signals for identical recent price action on different days.

This is exactly the class of train/serve skew that `tests/test_scale_consistency.py` was written to catch for the pickled sklearn models; the DQN's normalization is outside that contract because it lives in the env rather than in the artifact.

## Suggested fix

Persist the scaler with the artifact and use it verbatim at inference, the same contract the `.pkl` models already have via `predictors.base._scale`:

1. Thread `env.scaler` (or a single shared scaler fit across the training symbols) into `DQNAgent.save`'s blob, alongside a `feature_contract` matching `FEATURE_COLS`.
2. In `predict_symbol`, apply the loaded scaler instead of re-deriving mu/sd, and raise (rather than silently proceeding) if the artifact predates the change.
3. Pass `spy_df` into `TradingEnv`'s `make_daily_features` call so `ret_*_vs_spy` are real during training, or drop those two columns from the DQN's feature set — but not one convention in training and the other at inference.

A `test_scale_consistency.py`-style test asserting that the live DQN state vector for a fixed bar equals the env's observation for that same bar would lock this down.

---
*Auto-detected by codebase-review agent*

**Comment by seansorek** [comment]
Fixed by #125. Correcting state from 'not planned' to 'completed'.
